app/models/github_status.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+class GithubStatus
+  Status = Struct.new(:context, :latest_status) do
+    def state
+      latest_status.state
+    end
+
+    def description
+      latest_status.description
+    end
+
+    def url
+      latest_status.target_url
+    end
+
+    def success?
+      state == "success"
+    end
+
+    def failure?
+      state == "failure"
+    end
+
+    def pending?
+      state == "pending"
+    end
+  end
+
+  attr_reader :state, :statuses
+
+  def initialize(state, statuses)
+    @state = state
+    @statuses = statuses
+  end
+
+  def self.fetch(repo, ref)
+    cache_key = [name, repo, ref]
+
+    response = Rails.cache.read(cache_key)
+
+    # Fetch the data if the cache returned nil.
+    response ||= begin
+      GITHUB.combined_status(repo, ref)
+    rescue Octokit::Error
+      nil
+    end
+
+    # Fall back to a "missing" status.
+    return new("missing", []) if response.nil?
+
+    # Don't cache pending statuses, since we expect updates soon.
+    unless response&.state == "pending"
```

Comment:
don't need the `&.`since nil is handled above

Reply:
:+1: